catalog/core/catalog-core-versioning/versioning-common/src/main/java/ddf/catalog/core/versioning/impl/MetacardVersionImpl.java

```diff
@@ -183,44 +183,77 @@ public MetacardVersionImpl(String id, Metacard sourceMetacard, Action action, Su
         this.setTags(Collections.singleton(VERSION_TAG));
     }
 
+    //@formatter:off
     private byte[] getVersionType(Metacard sourceMetacard) {
-        try (ByteArrayOutputStream baos = new ByteArrayOutputStream();
-                ObjectOutputStream oos = new ObjectOutputStream(baos)) {
-            oos.writeObject(sourceMetacard.getMetacardType());
-            return baos.toByteArray();
-        } catch (IOException e) {
+        try {
+            return METACARD_TYPE_CACHE.get(sourceMetacard.getMetacardType()
+                    .hashCode(), () -> {
+                    try (ByteArrayOutputStream baos = new ByteArrayOutputStream();
+                        ObjectOutputStream oos = new ObjectOutputStream(baos)) {
```

Comment:
Though, for science I did confirm what would happen if only one were closed. 
If the Outer `ObjectOutputStream` was closed, it does automatically close the stream it wraps (`baos`). 
However if just the `ByteArrayOutputStream` were closed it would not close the outer `oos`.